.github/workflows/sycl_post_commit.yml

```diff
@@ -47,76 +47,18 @@ jobs:
-      with:
-        path: src
-    - name: Install Ubuntu deps
-      run: sudo apt install -y ninja-build ccache
-    - name: Setup Cache
-      uses: actions/cache@v3
-      id: cache
-      with:
-        path: ${{ github.workspace }}/cache
-        key: build-linux-shared-libs-${{ github.sha }}
-        restore-keys: |
-          build-linux-shared-libs-
-    - name: Configure
-      run: |
-        wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
-        sudo add-apt-repository "deb http://apt.llvm.org/focal/   llvm-toolchain-focal-15 main"
-        sudo apt-get update
-        sudo apt-get install -y clang-15
-        export CC="clang-15"
-        export CXX="clang++-15"
-        mkdir -p $GITHUB_WORKSPACE/build
-        cd $GITHUB_WORKSPACE/build
-        python3 $GITHUB_WORKSPACE/src/buildbot/configure.py -w $GITHUB_WORKSPACE \
-         -s $GITHUB_WORKSPACE/src -o $GITHUB_WORKSPACE/build -t Release \
-         --ci-defaults --shared-libs --cmake-opt="-DLLVM_CCACHE_BUILD=ON" \
-         --cmake-opt="-DLLVM_CCACHE_DIR=$GITHUB_WORKSPACE/cache" \
-         --cmake-opt="-DLLVM_CCACHE_MAXSIZE=2G"
-    - name: Compile
-      run: |
-        python3 $GITHUB_WORKSPACE/src/buildbot/compile.py -w  $GITHUB_WORKSPACE \
-         -s  $GITHUB_WORKSPACE/src -o $GITHUB_WORKSPACE/build
-    - name: check-llvm
-      if: always()
-      run: |
-        python3 $GITHUB_WORKSPACE/src/buildbot/check.py -w  $GITHUB_WORKSPACE \
-         -s  $GITHUB_WORKSPACE/src -o $GITHUB_WORKSPACE/build -t check-llvm
-    - name: check-clang
-      if: always()
-      run: |
-        python3 $GITHUB_WORKSPACE/src/buildbot/check.py -w  $GITHUB_WORKSPACE \
-         -s  $GITHUB_WORKSPACE/src -o $GITHUB_WORKSPACE/build -t check-clang
-    - name: check-sycl
-      if: always()
-      run: |
-        python3 $GITHUB_WORKSPACE/src/buildbot/check.py -w  $GITHUB_WORKSPACE \
-         -s  $GITHUB_WORKSPACE/src -o $GITHUB_WORKSPACE/build -t check-sycl
-    - name: check-llvm-spirv
-      if: always()
-      run: |
-        python3 $GITHUB_WORKSPACE/src/buildbot/check.py -w  $GITHUB_WORKSPACE \
-         -s  $GITHUB_WORKSPACE/src -o $GITHUB_WORKSPACE/build -t check-llvm-spirv
-    - name: check-xptifw
-      if: always()
-      run: |
-        python3 $GITHUB_WORKSPACE/src/buildbot/check.py -w  $GITHUB_WORKSPACE \
-         -s  $GITHUB_WORKSPACE/src -o $GITHUB_WORKSPACE/build -t check-xptifw
-    - name: Pack
-      run: tar -czvf llvm_sycl.tar.gz -C $GITHUB_WORKSPACE/build/install .
-    - name: Upload artifacts
-      uses: actions/upload-artifact@v3
-      with:
-        name: sycl_linux_${{ matrix.config }}
-        path: llvm_sycl.tar.gz
+    uses: ./.github/workflows/sycl_linux_build_and_test.yml
+    with:
+      build_cache_root: "/__w/llvm"
+      build_cache_suffix: sprod_shared
+      build_artifact_suffix: sprod_shared
+      build_configure_extra_args: --shared-libs
+      build_image: "ghcr.io/intel/llvm/sycl_ubuntu2004_nightly:no-drivers"
```

Comment:
Turns out the compiler from this image is broken (log - https://github.com/intel/llvm/actions/runs/4581226882/jobs/8090605649).
Fix - https://github.com/intel/llvm/pull/8926.